packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1357,7 +1590,105 @@ test.serial('writeGranuleFromApi() writes a granule to PostgreSQL, DynamoDB, and
+    esClient,
+    esGranulesClient,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+  } = t.context;
+
+  await writeGranuleFromApi({ ...granule }, knex, esClient, 'Create');
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  const originalpgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
+    status: granule.status,
+  };
+
+  const { updatedPgGranuleFields, pgGranule } = await updateGranule(t, updateGranulePayload);
+
+  // FUTURE:
+  // 1. 'created_at' is updated during PUT/PATCH
+  // 2. 'published' defaults to false if not provided in the payload
+  const omitList = ['cumulus_id', 'updated_at', 'created_at', 'published', 'timestamp'];
+
+  // Postgres granule matches expected updatedGranule
+  t.deepEqual(
+    omit(removeNilProperties(pgGranule), omitList),
+    omit(removeNilProperties({ ...originalpgGranule, ...updatedPgGranuleFields }), omitList)
+  );
+});
+
+test.serial('writeGranuleFromApi() given a partial granule updates all datastores with the same data', async (t) => {
+  const {
+    collectionCumulusId,
+    esClient,
+    esGranulesClient,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+  } = t.context;
+
+  await writeGranuleFromApi({ ...granule }, knex, esClient, 'Create');
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
+    status: granule.status,
+  };
+
+  const { esGranule, dynamoGranule, pgGranule } = await updateGranule(t, updateGranulePayload);
+
+  // Postgres and ElasticSearch granules matches
+  t.deepEqual(
+    omit(removeNilProperties(pgGranule), ['cumulus_id']),
+    await translateApiGranuleToPostgresGranule(esGranule, knex)
```

Comment:
I don't think this is quite right - api->postgres granule omits files.     To test this properly I think we need to translate the postgres granule to the API format as it's file inclusive.

Reply:
We sure do. See comment and updates below.